File: Level 0/코딩테스트 입문/평행/solution.py
def inclination(a, b):
    x1, y1 = a
    x2, y2 = b
    return (y2 - y1) / (x2 - x1)


def get_is_parrarel(a, b, c, d):
    return inclination(a, b) == inclination(c, d)


def solution(dots):
    a, b, c, d = dots

    case1 = get_is_parrarel(a, b, c, d)
    case2 = get_is_parrarel(a, c, b, d)
    case3 = get_is_parrarel(a, d, b, c)

    return 1 if True in [case1, case2, case3] else 0

File: Level 0/코딩테스트 입문/평행/solution.js
function solution(dots) {
  const [a, b, c, d] = dots;

  const case1 = getIsParrarel(a, b, c, d);
  const case2 = getIsParrarel(a, c, b, d);
  const case3 = getIsParrarel(a, d, b, c);

  return [case1, case2, case3].includes(true) ? 1 : 0;
}

function getInclination(a, b) {
  const [x1, y1] = a;
  const [x2, y2] = b;
  return (y2 - y1) / (x2 - x1);
}

function getIsParrarel(a, b, c, d) {
  return getInclination(a, b) === getInclination(c, d);
}
